test/fixtures/require-resolve.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const path = require('path');
+const nodeModules = path.join(__dirname, 'node_modules');
+const nestedNodeModules = path.join(__dirname, 'node_modules', 'node_modules');
+const nestedIndex = path.join(__dirname, 'nested-index');
+
+// Test the default behavior.
+assert.strictEqual(
+  require.resolve('bar'),
+  path.join(nodeModules, 'bar.js')
+);
+
+// Verify that existing paths are removed.
+assert.throws(() => {
+  require.resolve('bar', { paths: [] })
+}, /^Error: Cannot find module 'bar'$/);
+
+// Verify that resolution path can be overwritten.
+{
+  // three.js cannot be loaded from this file by default.
+  assert.throws(() => {
+    require.resolve('three')
+  }, /^Error: Cannot find module 'three'$/);
+
+  // However, it can be found if resolution contains the nested index directory.
+  assert.strictEqual(
+    require.resolve('three', { paths: [nestedIndex] }),
+    path.join(nestedIndex, 'three.js')
+  );
+
+  // Resolution from nested index directory also checks node_modules.
+  assert.strictEqual(
+    require.resolve('bar', { paths: [nestedIndex] }),
+    path.join(nodeModules, 'bar.js')
```

Comment:
something wrong here? bar.js is not present under `nestedIndex` so can you double-check this check?
This is what I get for bar.js and three.js:
```
#find . -name bar.js
./test/fixtures/node_modules/bar.js
./test/fixtures/node_modules/node_modules/bar.js

#find . -name three.js
./test/fixtures/json-with-directory-name-module/module-stub/one/two/three.js
./test/fixtures/json-with-directory-name-module/module-stub/one-trailing-slash/two/three.js
./test/fixtures/nested-index/three.js
#
```

Reply:
The `require()` machinery checks the `node_modules` directory in the parent, which is how it finds `bar.js`.

Reply:
ah, ok - got it, thanks!